### PR TITLE
Storage fixups

### PR DESCRIPTION
**Pull request by gctucker**
Fix and improve a few things in `kernelci.config.storage` and `kernelci.storage`:
* fix docstrings
* fix YAML tags to match the class names
* convert Storage class to ABC
* add `._connect()` method
* allow `._upload()` method to return the URL mapping to override the default implementation

### CODE REVIEW

**Review by JenySadadia — CHANGES_REQUESTED** [review]
Maybe you could fix the module docstring here as well https://github.com/kernelci/kernelci-core/blob/main/kernelci/cli/storage.py#L6

File: kernelci/storage/__init__.py [hunk 1]
```diff
@@ -33,13 +34,23 @@ def credentials(self):
         """Credentials data"""
         return self._credentials
 
+    def _connect(self):
```
Comment (JenySadadia) [inline kernelci/storage/__init__.py:37]: This should also be declared as an abstract method.
```suggestion
   @abc.abstractmethod
    def _connect(self):
```

File: kernelci/config/storage.py [hunk 2]
```diff
@@ -11,6 +11,8 @@
 class Storage(YAMLConfigObject):
     """Base configuration class for Storage implementations"""
 
+    yaml_tag = '!Storage'
```
Comment (JenySadadia) [inline kernelci/config/storage.py:14]: Where is `yaml_tag` being used?

Comment (gctucker) [reply kernelci/config/storage.py:14]: Every time a base class `Storage` config object get deserialised from YAML.  All the current storage types defined in the file have inherited classes but it's entirely possible to have a storage type that just relies on the base Storage config.

Comment (gctucker) [reply kernelci/config/storage.py:14]: See also: https://pyyaml.org/wiki/PyYAMLDocumentation

File: kernelci/storage/__init__.py (hunk 1 above)
Comment (gctucker) [reply kernelci/storage/__init__.py:37]: No because it doesn't need to be implemented.

**Comment by gctucker** [comment]
> Maybe you could fix the module docstring here as well https://github.com/kernelci/kernelci-core/blob/main/kernelci/cli/storage.py#L6

Right, that's a CLI tool issue rather than storage.  But yes as this PR is a bit of a catch-all one for any storage-related fixes so I'll add a commit for it.

File: kernelci/storage/__init__.py (hunk 1 above)
Comment (JenySadadia) [reply kernelci/storage/__init__.py:37]: Yes, I saw that it is not implemented in `StorageBackend`.

**Comment by gctucker** [comment]
@JenySadadia Please take another look, I believe I've addressed all your feedback.

**Review by JenySadadia — APPROVED** [review]
LGTM

**Pull request merged**